contracts/Whitelist.sol

```diff
@@ -105,60 +116,99 @@ contract Whitelist is Ownable {
+     * @param _collateral collateral asset address
+     */
+    function blacklistCollateral(address _collateral) external onlyOwner {
+        _setWhitelistedCollateral(_collateral, false);
+
+        emit CollateralBlacklisted(_collateral);
+    }
+
     /**
      * @notice allow Otoken Factory to whitelist a new option
-     * @dev can only be called from the owner's address
+     * @dev can only be called from the Otoken Factory address
      * @param _otokenAddress otoken
      */
     function whitelistOtoken(address _otokenAddress) external onlyFactory {
-        _setWhitelistedOtoken(_otokenAddress);
+        _setWhitelistedOtoken(_otokenAddress, true);
 
         emit OtokenWhitelisted(_otokenAddress);
     }
 
     /**
-     * @notice set a product hash as whitelisted
-     * @param _productHash product hash in bytes
+     * @notice allow owner to blacklist a new option
+     * @dev can only be called from the owner's address
+     * @param _otokenAddress otoken
      */
-    function _setWhitelistedProduct(bytes32 _productHash) internal {
-        require(!whitelistedProduct[_productHash], "Whitelist: Product already whitelisted");
+    function blacklistOtoken(address _otokenAddress) external onlyOwner {
+        _setWhitelistedOtoken(_otokenAddress, false);
 
-        whitelistedProduct[_productHash] = true;
+        emit OtokenBlacklisted(_otokenAddress);
+    }
+
+    /**
+     * @notice set product state
+     * @param _product product hash in bytes
+     * @param _state represent if a product is whitelisted or not, true to whitelist and false to blacklist
+     */
+    function _setWhitelistedProduct(bytes32 _product, bool _state) internal {
+        whitelistedProduct[_product] = _state;
     }
 
     /**
      * @notice set a collateral address as whitelisted
      * @param _collateral collateral address
+     * @param _state represent if a collateral is whitelisted or not, true to whitelist and false to blacklist
      */
-    function _setWhitelistedCollateral(address _collateral) internal {
-        require(!whitelistedCollateral[_collateral], "Whitelist: Collateral already whitelisted");
-
-        whitelistedCollateral[_collateral] = true;
+    function _setWhitelistedCollateral(address _collateral, bool _state) internal {
+        whitelistedCollateral[_collateral] = _state;
     }
 
     /**
      * @notice set an otoken address as whitelisted
      * @param _otokenAddress address of the oToken that is being whitelisted
+     * @param _state represent if a Otoken is whitelisted or not, true to whitelist and false to blacklist
      */
-    function _setWhitelistedOtoken(address _otokenAddress) internal {
-        require(!whitelistedOtoken[_otokenAddress], "Whitelist: Otoken already whitelisted");
-
-        whitelistedOtoken[_otokenAddress] = true;
+    function _setWhitelistedOtoken(address _otokenAddress, bool _state) internal {
+        whitelistedOtoken[_otokenAddress] = _state;
```

Comment:
I don't think we need this function anymore if it's just `whitelistedOtoken[_otokenAddress] = _state;`

Reply:
yeah agree, removed that.